tests/apis.js

```diff
@@ -52,80 +51,300 @@ test.cb.after.always(t => {
+    if (attr.split('-').indexOf('referencer') > -1) {
+      const item = attrs[attr];
+
+      // if item has attributes
+      if (item.hasOwnProperty('attributes')) {
+        referencerTests(t, item.attributes);
+
+        return;
+      }
+
+      // if it's an array, recurseit!
+      if (Array.isArray(item)) {
+        referencerTests(t, item);
+
+        return;
+      }
+
+      if (typeof item === 'object' && typeof item[Object.keys(item)[0]] === 'object' && item[Object.keys(item)[0]].hasOwnProperty('attributes')) {
+        referencerTests(t, item.attributes);
+
+        return;
+      }
+
+      if (Object.keys(item).indexOf('id') === -1 && Object.keys(item).length >= 1) {
+        Object.keys(item).forEach(itm => {
+          t.true(item[itm].hasOwnProperty('meta'), 'attribute in array contains meta');
+
+          return;
+        });
+      }
+
+      // non-repeatable; single input
+      else {
+        t.true(item.hasOwnProperty('meta'), 'attribute in array contains meta');
+
+        return;
+      }
+    }
+  });
+};
+
 //////////////////////////////
 // Utils - attributes
 //////////////////////////////
-test('Utils: attributes', t => {
-  const item = Math.round(Math.random() * (content.length - 1));
-  let expected = content[item];
+test.serial.skip('Utils: attributes - empty query', t => {
+  const random = Math.round(Math.random() * live.length - 1);
+  let expected = live[random];
   if (expected === undefined) {
-    expected = cloneDeep(content[(content.length - 1)]);
+    expected = cloneDeep(live[(live.length - 1)]);
   }
 
   const model = allTypes.find(typ => {
     return typ.id === expected['type-slug'];
   });
 
-  const attributes = apiUtils.attributes(expected.value, model.attributes);
+  const query = {};
 
-  t.is(typeof attributes, 'object', 'Should return an object.');
-  Object.keys(attributes).map(key => {
-    const attr = key.split('-');
-    if (attr[attr.length - 1] === 'referencer') {
-      if (attributes[key] !== '') {
-        t.true(isUUID(attributes[key]), 'includes a uuid');
-      }
+  const attributes = apiUtils.attributes(expected.attributes, model.attributes, allTypes, query);
+
+  return attributes.then(result => {
+    t.is(typeof result, 'object', 'Should contain result, an object.');
+    t.is(Object.keys(result).length, 6, 'Should contain six main objects.');
+
+    referencerTests(t, result);
+  })
+  .catch(e => {
+    console.log(e); // eslint-disable-line no-console
+  });
+});
+
+test.serial.skip('Utils: attributes - depth 0', t => {
+  const random = Math.round(Math.random() * live.length - 1);
+  let expected = live[random];
+  if (expected === undefined) {
+    expected = cloneDeep(live[(live.length - 1)]);
+  }
+
+  const model = allTypes.find(typ => {
+    return typ.id === expected['type-slug'];
+  });
+
+  // because depth is checked during attributes, depth 1 and 0 are the same for the attributes function
+  const query = {
+    depth: 0,
```

Comment:
I'm confused, this comment is not true. `depth=0` should test to make sure that references returned have their `meta` information, whereas `depth=1` should check to make sure that references returned have their full attribute information, and any references there should have their `meta` information